packages/frames/src/Frame.ts

```diff
@@ -0,0 +1,186 @@
+import { MetaTag } from 'next-seo/lib/types';
+
+export enum FrameAspectRatio {
+  SQUARE = '1:1',
+  WIDE = '1.91:1',
+}
+
+export enum FrameButtonAction {
+  /**
+   * The button will trigger a post request to the postUrl.
+   * By default, the postUrl is used as the target URL if target is not provided.
+   */
+  POST = 'post',
+  /**
+   * The button will redirect to the target URL.
+   */
+  LINK = 'link',
+}
+
+export interface FrameButton {
+  /**
+   * The label to be displayed in the button.
+   */
+  label: string;
+  /**
+   * - post: The button will trigger a post request to the postUrl.
+   * - link: The button will redirect to the target URL.
+   * @default post
+   */
+  action?: FrameButtonAction;
+  /**
+   * A target URL for the button. It uses the postUrl if not provided.
+   */
+  target?: string;
+}
+
+interface FrameConstructorProps {
+  /**
+   * The title to be displayed in the frame. Required.
+   */
+  title: string;
+  /**
+   * The URL of the image to be displayed in the frame. It should be a URL. Required.
+   */
+  image: string;
+  /**
+   * The URL to be used in the post request. Required.
+   */
+  postUrl: string;
+  /**
+   * Aspect-ratio of the image. Required.
+   */
+  aspectRatio: FrameAspectRatio;
+  /**
+   * An ordered list of buttons to be displayed in the frame.
+   */
+  buttons?: FrameButton[];
+}
+
+/**
+ * Represents a Farcaster frame.
+ * Documentation: https://docs.farcaster.xyz/learn/what-is-farcaster/frames
+ * Specification: https://docs.farcaster.xyz/reference/frames/spec
+ */
+export class Frame {
+  /**
+   * The title to be displayed in the frame. Required.
+   */
+  title: string;
+  /**
+   * The URL of the image to be displayed in the frame. It should be a URL. Required.
+   */
+  image: string;
+  /**
+   * The URL to be used in the post request. Required.
+   */
+  postUrl: string;
+  /**
+   * Aspect-ratio of the image. Required.
+   */
+  aspectRatio: FrameAspectRatio;
+  /**
+   * An ordered list of buttons to be displayed in the frame.
+   */
+  buttons?: FrameButton[];
+
+  constructor(props: FrameConstructorProps) {
+    this.title = props.title;
+    this.image = props.image;
+    this.postUrl = props.postUrl;
+    this.aspectRatio = props.aspectRatio;
+    this.buttons = props.buttons;
```

Comment:
Maybe some validations could be done over these.

For example:

> The type of image must be jpg, png or gif.

> Sanitize redirect URLs to ensure they start with http:// or https://.

Maybe it is not worth it, is just a suggestion to contemplate.

Reply:
Regarding the image type:

I think the only way of doing it's by making a `fetch HEAD` request of the file URL and checking the Content-Type. And let assume the image does not comply with the supported formats, what we do next? should we use a placeholder instead? 

Maybe it make sense having some fallback image, yesterday we were testing the Moment Frame and someone upload a HEIC image and the Frame explodes when displaying the image.

Let me know what you think.

Reply:
Validating images, since those can be user-content (like Moments), should be validated if we can do it in a timely manner and avoid delaying frames too much.

It could also be configurable. Some apps do control what image is displayed in the frame, and the extra time needed to validate the image is not necessary.

Reply:
I like the idea of a fallback; I would make it an instance parameter and use a POAP default image if I am not given one. So we can change it for each project.

Reply:
If goes further, because it can also get `data:` URIs with the image data in base64.

Reply:
I suggest we move forward with this for version `1.0.0`. I will create an issue with this discussion.